experimental/python/codegen/codegen/generated_dataclass_patch.py

```diff
@@ -0,0 +1,153 @@
+    It's necessary for kwargs in the constructor to work correctly.
+    """
+    for name, model in models.items():
+        if not model.fields:
+            continue
+
+        required_fields = [field for field in model.fields if _is_required(field)]
+        optional_fields = [field for field in model.fields if not _is_required(field)]
+
+        models[name] = replace(model, fields=required_fields + optional_fields)
+
+
+def quote_recursive_references(models: dict[str, GeneratedDataclass]):
+    """
+    If there is a cycle between two dataclasses, we need to quote one of them.
+
+    Example:
+
+        class Foo:
+            bar: Optional[Bar]
+
+        class Bar:
+            foo: "Foo"
+    """
+
+    # see also _append_resolve_recursive_imports
+
+    models["jobs.ForEachTask"] = _quote_recursive_references_for_model(
+        models["jobs.ForEachTask"],
+        references={"Task", "TaskParam"},
+    )
+
+
+def _quote_recursive_references_for_model(
+    model: GeneratedDataclass,
+    references: set[str],
+) -> GeneratedDataclass:
+    def update_type_name(type_name: GeneratedType):
+        if type_name.name in references:
+            return replace(
+                type_name,
+                name=f'"{type_name.name}"',
+            )
+        elif type_name.parameters:
+            return replace(
+                type_name,
+                parameters=[update_type_name(param) for param in type_name.parameters],
+            )
+        else:
+            return type_name
+
+    def update_field(field: GeneratedField):
+        return replace(
+            field,
+            type_name=update_type_name(field.type_name),
+            param_type_name=update_type_name(field.param_type_name),
+        )
+
+    return replace(
+        model,
+        fields=[update_field(field) for field in model.fields],
+    )
+
+
+def _is_required(field: GeneratedField) -> bool:
+    return field.default is None and field.default_factory is None
+
+
+def add_default_values(models: dict[str, GeneratedDataclass]):
+    models["jobs.CronSchedule"] = _add_default_value(
+        models["jobs.CronSchedule"],
+        field_name="timezone_id",
+        default_value='"UTC"',
+    )
+
+
+def add_oneofs(models: dict[str, GeneratedDataclass]):
+    models["jobs.JobRunAs"] = _add_oneof(
+        models["jobs.JobRunAs"],
+        ["user_name", "service_principal_name"],
+        required=True,
+    )
+
+    models["resources.Permission"] = _add_oneof(
+        models["resources.Permission"],
+        ["user_name", "service_principal_name", "group_name"],
+        required=True,
+    )
+
+    models["jobs.Task"] = _add_oneof(
+        models["jobs.Task"],
+        ["new_cluster", "job_cluster_key", "environment_key", "existing_cluster_id"],
+        required=False,
+    )
+
+    models["jobs.TriggerSettings"] = _add_oneof(
+        models["jobs.TriggerSettings"],
+        ["file_arrival", "periodic", "table_update"],
+        required=True,
+    )
```

Comment:
Can we codify this upstream?

Reply:
Created https://github.com/databricks/cli/pull/2670, I think the same argument applies for default values, they aren't part of OpenAPI spec right now.